cmd/frontend/internal/bg/migrate_saved_queries_to_db.go

```diff
@@ -0,0 +1,175 @@
+package bg
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/db"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/types"
+	"github.com/sourcegraph/sourcegraph/pkg/api"
+	"github.com/sourcegraph/sourcegraph/pkg/db/dbconn"
+	"github.com/sourcegraph/sourcegraph/pkg/jsonc"
+	log15 "gopkg.in/inconshreveable/log15.v2"
+)
+
+type SavedQuery struct {
+	Key         string
+	Description string
+	Query       string
+	Notify      bool
+	NotifySlack bool
+}
+
+type SavedQueryField struct {
+	SavedQueries []SavedQuery `json:"search.savedQueries"`
+}
+
+// MigrateSavedQueriesAndSlackWebhookURLsFromSettingsToDatabase migrates saved searches from the `search.SavedQueries` value in
+// site settings into the `saved_searches` PostgreSQL table, and migrates organization Slack webhook URLs from the
+// `notifications.slack` value in site settings to the `slack_webhook_url` column in the `saved_searches` table.
+//
+// This migration does NOT remove the existing "search.savedQueries" and "notifications.slack" values from the user or org's settings.
+// This is to be done in a future iteration after all customer instances have upgraded and ran the migration successfully.
+func MigrateSavedQueriesAndSlackWebhookURLsFromSettingsToDatabase(ctx context.Context) {
+	savedSearchTableIsEmpty, err := db.SavedSearches.IsEmpty(ctx)
+	if err != nil {
+		log15.Error(`Warning: unable to migrate search.savedQueries because cannot tell if saved searches table is empty. Please report this issue.`, err)
+	}
+	if savedSearchTableIsEmpty {
+		if err := doMigrateSavedQueriesAndSlackWebhookURLsFromSettingsToDatabase(ctx); err != nil {
+			log15.Error(`Warning: unable to migrate "search.savedQueries" settings to database. Please report this issue.`, "error", err)
+		}
+	}
+}
+
+func doMigrateSavedQueriesAndSlackWebhookURLsFromSettingsToDatabase(ctx context.Context) error {
+	// List all settings that include a `search.savedQueries` field.
+	settings, err := db.Settings.ListAll(ctx, "search.savedQueries")
+	if err != nil {
+		// Don't continue the migration if we can't list settings.
+		return errors.WithMessagef(err, `Warning: unable to migrate saved queries to database. Error listing settings. Please report this issue`)
```

Comment:
```suggestion
		return errors.WithMessagef(err, "Warning: unable to migrate saved queries to database. Error listing settings. Please report this issue")
```

Only use raw string when needed (i.e. when there are double quotes or newlines in the string)